docs/proposals/mapper- provide-a- common- MQTT- mapper.md

```diff
@@ -0,0 +1,277 @@
+
+        Firstly, different device manufacturers may implement customizations and extensions to the MQTT protocol, leading to interoperability issues between devices. Secondly, the diversity of serialization formats presents another challenge when handling device data. JSON, YAML, and XML are common serialization methods, each with its unique advantages and application scenarios.
+
+        Therefore, developing a universal MQTT can help developers simplify the device integration process, improve system flexibility and reliability.
+
+### Goals
+
+        For developers or end users of a common MQTT mapper solutions, the goals of a common MQTT mapper are:
+
+- Add a new Mapper for the MQTT protocol to the kubeedge/mappers-go project using the latest mapper-framework.
+
+- Mapper can get the message from the specified MQTT service's topic, and can parse the attribute values from the message using common serialization methods(JSON, YAML, XML). 
+
+- Provide a more scalable solution for parsing attribute values from more customized messages wherever possible.
+
+### Proposal
+
+        Providing a common MQTT mapper using the latest mapper-framework aims to help developers easily make use of mapper to solve MQTT communication, in order to simplify the device integration process, improve system flexibility and reliability.
+
+##### Scope
+
+- Development of the MQTT Mapper: Utilize the latest Mapper-Framework to add an MQTT protocol Mapper to the kubeedge/mappers-go project. Ensure the Mapper can subscribe to specified MQTT topics, receive messages, and parse them.
+
+- Message Parsing and Serialization Support: functionality to retrieve messages from MQTT service topics. Support parsing messages using JSON, YAML and XML formats to extract attribute values.
+
+##### User
+
+- Developers: Developers are needed to implement and test the MQTT Mapper.
+
+- IoT Device Manufacturers and Operators: IoT device manufacturers and operators can use this universal MQTT Mapper to manage device communications.
+
+### Design Details
+
+##### Architecture and Modules
+
+        The architecture for Mapper Provide a Common Mapper will be modular to ensure scalability and maintainability. The main components include:
+
+- Base Environment Setup: Utilizes the latest mapper-framework to create a MQTT mapper for standardized operatons and interactions.
+
+- Main algorithm design: Parses the attribute values from the message using common serialization methods(JSON, YAML, XML).
+
+        The architectures and related concepts are shown in the below figure. In the figure, we assume that in the cloud-side-end scenario, device state modifications can be initiated by the cloud.
+
+<img title="相对路径演示" src="../images/proposals/mapper-provide-a-common-MQTT-mapper-architecture.jpg" alt="本地路径" width="546" data-align="center">
```

Comment:
The communication method between devicetwin and mapper in the architecture diagram is wrong. Now it relies on dmi in devicetwin and defines the grpc interface for implementation, rather than mqtt broker.

![edge](https://github.com/user-attachments/assets/bb783d37-94c6-4160-8803-4d0253107371)

Reply:
Ok, I will use the architecture diagram you provided me as an overall description of Kubeedge.